### PR TITLE
Proration and credit calculated amounts

### LINKED ISSUE #1869: Display any applicable credit
When changing plans, any credits applicable should be displayed to the user on the `Confirm plan change` slide of the `Change Plan` modal dialog, between the rest of the content and the total value. The total displayed should reflect the amount the user will be charged now. The information can be obtained from `ApiClient.checkSubscriptionUpdate(...)` response.

image

### PR DESCRIPTION
**Pull request by tanmoyAtb**
closes #1869

---
Submission checklist: 

   

#### Layout
- [x] Change looks good in the desktop web ui
- [x] Change looks good in the mobile web ui

#### Theme
- [x] Components / elements inspected in light mode 
- [x] Components / elements inspected in dark mode

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx [hunk 1]
```diff
@@ -114,12 +115,22 @@ const ConfirmPlan = ({
...
-  }, [currentSubscription, planPrice]
-  )
+  }, [currentSubscription, planPrice])
+
+  useEffect(() => {
+    if (!currentSubscription) return
+    filesApiClient.checkSubscriptionUpdate(currentSubscription?.id, {
+      payment_method: paymentMethod === "creditCard" ? "stripe" : "crypto",
+      price_id: planPrice.id
+    }).then(setCheckSubscriptionUpdate).catch(console.error)
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx]: some air
```suggestion
    if (!currentSubscription) return

    filesApiClient.checkSubscriptionUpdate(currentSubscription?.id, {
      payment_method: paymentMethod === "creditCard" ? "stripe" : "crypto",
      price_id: planPrice.id
    })
    .then(setCheckSubscriptionUpdate)
    .catch(console.error)
```

**Review by Tbaut — COMMENTED** [review]
I went briefly though the code, lgtm, I'll play with it now.

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx [hunk 2]
```diff
@@ -278,6 +286,36 @@ const ConfirmPlan = ({
...
+              component="h5"
+              className={classes.boldText}
+            >
+              {planPrice.currency} {checkSubscriptionUpdate.amount_remaining.toFixed(2)}
+            </Typography>
+            {checkSubscriptionUpdate.amount_balance > 0 && <Typography
+              component="p"
+              variant="body2"
+            >
+              (<Trans>Previous balance</Trans> : {planPrice.currency} {checkSubscriptionUpdate.amount_balance.toFixed(2)})
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx]:
```suggestion
              (<Trans>Previous balance</Trans>: {planPrice.currency} {checkSubscriptionUpdate.amount_balance.toFixed(2)})
```

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx [hunk 3]
```diff
@@ -278,6 +286,36 @@ const ConfirmPlan = ({
...
+            </Typography>
+
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx]:
```suggestion
```

File: packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx [hunk 4]
```diff
@@ -299,7 +337,7 @@ const ConfirmPlan = ({
           <Button
             variant="primary"
             loading={loadingChangeSubscription}
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/Settings/SubscriptionTab/ChangePlan/ConfirmPlan.tsx]: I'd add it here as well, wdyt?
```suggestion
            loading={loadingChangeSubscription || !checkSubscriptionUpdate}
```

**Review by Tbaut — APPROVED** [review]
Looking good, it's simple and does the job well IMHO! Left some more nits.

**Comment by asnaith** [comment]
This seems to remember only the last payment when performing the calculations. Is this how it should work? I'm finding it a bit confusing so I'm sure users will too. 

I'm not sure this is a genuine flow that someone would do but currently - 

If I purchase a premium plan for $199 then if I downgrade to a standard plan then I don't owe anything because I paid more for the premium. However, if I then try to upgrade the standard plan back to premium I will have to pay an additional $100 as the standard account was only $99...but really the account should still have another $100 associated with it and no cost to go back to premium because the initial payment was $199.  

Similarly, I purchase an annual standard plan for $99 and then I switch to a monthly premium plan the cost is $0.00 but it's not clear to me if the previously paid $99 will mean I don't get billed for the first 5 months when on the premium plan. 

Hope this makes sense!

**Comment by Tbaut** [comment]
I had this thought too regarding the fact that we don't say anything about future invoices that might be impacted by the current account balance, but I figured this is something that the user intuitively understands, when telling them how much their account balance is.

I had a somewhat similar "issue" that my balance seemed to have been reset when switching to the free plan, but I guess this might be something to discuss with the api team.

**Comment by tanmoyAtb** [comment]
@asnaith Thanks for bringing this up. I noticed the first flow mismatch as well. However the calculations are fully done on the API side. Can you post the two flows on the files channel and we can discuss it with the team. 

also we're only getting the next payment amounts from the API. IMO thats sufficient for the user to understand that the next few payments will be adjusted.

**Review by FSM1 — CHANGES_REQUESTED** [review]
Works great. The only thing I would change here is some of the labels - instead of 
image

I would display the following:
```
Plan price: $ 19.99
Credit used: $9.99
--------------------
Payment Due: $10.00
```

**Comment by tanmoyAtb** [comment]
> Works great. The only thing I would change here is some of the labels - instead of image
> 
> I would display the following:
> 
> ```
> Plan price: $ 19.99
> Credit used: $9.99
> --------------------
> Payment Due: $10.00
> ```

@FSM1 do you mean `credit used` or `credit remaining` ?

**Comment by FSM1** [comment]
> > Works great. The only thing I would change here is some of the labels - instead of image
> > I would display the following:
> > ```
> > Plan price: $ 19.99
> > Credit used: $9.99
> > --------------------
> > Payment Due: $10.00
> > ```
> 
> @FSM1 do you mean `credit used` or `credit remaining` ?

I was just thinking of that... how about we go for something like the:

```
Plan price: $ 19.99
Credit used: $9.99
--------------------
Payment Due: $10.00
Credit remaining: $ 0 (Only display if there is still credit remaining - i.e. Payment Due === 0
```

**Comment by tanmoyAtb** [comment]
> > > Works great. The only thing I would change here is some of the labels - instead of image
> > > I would display the following:
> > > ```
> > > Plan price: $ 19.99
> > > Credit used: $9.99
> > > --------------------
> > > Payment Due: $10.00
> > > ```
> > 
> > 
> > @FSM1 do you mean `credit used` or `credit remaining` ?
> 
> I was just thinking of that... how about we go for something like the:
> 
> ```
> Plan price: $ 19.99
> Credit used: $9.99
> --------------------
> Payment Due: $10.00
> Credit remaining: $ 0 (Only display if there is still credit remaining - i.e. Payment Due === 0
> ```

Looks relatable to me, I'll update this once API makes the changes for `credit remaining` and updates the field names.

**Comment by tanmoyAtb** [comment]
Hey guys, The PR is good to be reviewed IMO. As long as the terminology looks relatable to you, Its good to go.

**Review by FSM1 — APPROVED** [review]

**Pull request merged**